### PR TITLE
Improve events

### PR DESCRIPTION
**Pull request by mdmohsin7**
Should close: #1100 
- [x] Fetch calendars twice with delay (because it does not always return all the calendars in one attempt most of the time)
Screenshot 2024-10-19 at 10 25 30 PM

- [x] Mention about events adding delay
- [x] Pass the user's timezone to GPT and force it to always return events in UTC (previously it was assuming the time mentioned by user in transcript to be in UTC)
- [x] Remove utc offset addition and convert UTC to local timezone on the front
- [x] Minor UI improvements

## Summary by CodeRabbit

- **New Features**
	- Enhanced user feedback with updated SnackBar message after adding an event to the calendar.
	- Improved clarity in shared content when using the sharing options.
	- Adjusted event start time handling to reflect local time for better user experience.
	- Added a method to retrieve user time zone for improved memory processing.

- **Bug Fixes**
	- Implemented a delay for calendar access requests to ensure calendars are fetched correctly.
	- Adjusted subtitle formatting for better readability in the calendar settings.

- **Documentation**
	- Updated comments for clarity regarding the new delay mechanism in calendar access handling.

### CODE REVIEW

File: backend/utils/memories/process_memory.py [hunk 1]
```diff
@@ -52,15 +52,16 @@ def _get_structured(
             return summarize_open_glass(memory.photos), False
 
         # from Friend
+        tz = notification_db.get_user_time_zone(uid)
         if force_process:
             # reprocess endpoint
-            return get_transcript_structure(memory.get_transcript(False), memory.started_at, language_code), False
+            return get_transcript_structure(memory.get_transcript(False), memory.started_at, language_code,tz), False
```
Comment (beastoin) [inline backend/utils/memories/process_memory.py:58]: pls check all usages of func `get_transcript_structure` in the backend side, ensure all of them uptodate with new params: tz of tz should be an optional param?

File: app/lib/providers/calendar_provider.dart [hunk 2]
```diff
@@ -45,23 +45,20 @@ class CalenderProvider extends ChangeNotifier {
         setLoading(true);
 
         await _getCalendars();
-
-        if (calendars.isEmpty) {
-          // try to get calendars again
+        // try to get calendars again after 3 seconds
+        // Delay is necessary because the calendar plugin does not return the calendars immediately
+        // While testing, on first call I got only 1 calendar, but on the second call I got all the calendars
```
Comment (beastoin) [inline app/lib/providers/calendar_provider.dart:50]: ah ha, send them and issue pls :)

File: app/lib/pages/memory_detail/widgets.dart [hunk 3]
```diff
@@ -206,7 +206,7 @@ class EventsListWidget extends StatelessWidget {
                   subtitle: Padding(
                     padding: const EdgeInsets.only(top: 4.0),
                     child: Text(
-                      '${dateTimeFormat('MMM d, yyyy', event.startsAt)} at ${dateTimeFormat('h:mm a', event.startsAt)} ~ ${minutesConversion(event.duration)}.',
+                      '${dateTimeFormat('MMM d, yyyy', utcToLocal(event.startsAt))} at ${dateTimeFormat('h:mm a', utcToLocal(event.startsAt))} ~ ${minutesConversion(event.duration)}',
```
Comment (beastoin) [inline app/lib/pages/memory_detail/widgets.dart]: toLocal should be done at the schema parsing(fromJson) ? if not pls put the toLocal code there. just like this:

Screenshot 2024-10-20 at 04 53 58

File: backend/utils/memories/process_memory.py (hunk 1 above)
Comment (mdmohsin7) [reply backend/utils/memories/process_memory.py:58]: There are multiple usages of `get_transcript_structure` (which mostly are old funcs), confirmed with Joan that only v2/memories endpoint is being used for memories. Anyway will update all of them just to be on the safe side.

tz is required for events to be in correct timezone, and we can never be sure if a conversation will contain an event or not before calling `get_transcript_structure`, so made it mandatory

Edit: Forgot to pass it in some places in v2/memories as well. fixing

File: app/lib/pages/memory_detail/widgets.dart (hunk 3 above)
Comment (mdmohsin7) [reply app/lib/pages/memory_detail/widgets.dart]: Oh yes totally forgot, let me do it

File: backend/utils/llm.py [hunk 4]
```diff
@@ -94,13 +94,15 @@ def get_transcript_structure(transcript: str, started_at: datetime, language_cod
         'format_instructions': parser.get_format_instructions(),
         'language_code': language_code,
         'started_at': started_at.isoformat(),
+        'tz': tz,
     })
+    print(response)
```
Comment (beastoin) [inline backend/utils/llm.py]: remove this log mannnn

Comment (mdmohsin7) [reply backend/utils/llm.py]: shit guess I forgot to remove after testing

**Comment by beastoin** [comment]
lgtm 🥳

**Pull request merged**